src/Filesystem/Provider/FileAccess.php

```diff
@@ -0,0 +1,257 @@
+<?php
+
+namespace ILIAS\Filesystem\Provider;
+
+use ILIAS\Filesystem\DTO\FileSize;
+use ILIAS\Filesystem\Exception\FileAlreadyExistsException;
+use ILIAS\Filesystem\Exception\FileNotFoundException;
+use ILIAS\Filesystem\Exception\IOException;
+
+/**
+ * Interface FileAccess
+ *
+ * The FileAccess interface defines all file operations.
+ * Filesystem role interface.
+ *
+ * @author  Nicolas Schäfli <ns@studer-raimann.ch>
+ * @since 5.3
+ * @version 1.0
+ *
+ * @public
+ */
+interface FileAccess {
+
+	/**
+	 * Writes the content to a new file.
+	 *
+	 * @param string $path      The path to the file which should be created.
+	 * @param string $content   The content which should be written to the new file.
+	 *
+	 * @return void
+	 *
+	 * @throws FileAlreadyExistsException   If the file already exists.
+	 * @throws IOException                  If the file could not be created or written.
+	 *
+	 * @since 5.3
+	 * @version 1.0
+	 */
+	public function write($path, $content);
+
+
+	/**
+	 * Updates the content of a file.
+	 * Replaces the file content with a new one.
+	 *
+	 * @param string $path          The path to the file which should be updated.
+	 * @param string $newContent    The new file content.
+	 *
+	 * @return void
+	 *
+	 * @throws FileNotFoundException    If the file is not found.
+	 * @throws IOException              If the file could not be updated.
+	 *
+	 * @since 5.3
+	 * @version 1.0
+	 */
+	public function update($path, $newContent);
```

Comment:
@klees Yes that could be helpful, but in terms of the logging we could implement a FileStreamHandler for the monolog library which allows us to stream the log to the corresponding log file.  [Example StreamHandler](https://github.com/Seldaek/monolog/blob/master/src/Monolog/Handler/StreamHandler.php)